src/ZODB/MappingStorage.py

```diff
@@ -269,9 +267,9 @@ def tpc_begin(self, transaction, tid=None):
         if transaction is self._transaction:
             raise ZODB.POSException.StorageTransactionError(
                 "Duplicate tpc_begin calls for same transaction")
-        self._lock_release()
+        self._lock.release()
         self._commit_lock.acquire()
-        self._lock_acquire()
+        self._lock.acquire()
```

Comment:
``utils.RLock`` is only used when debugging locks. The locks used here are regular locks normally.

This would be a little clearer if I didn't use the decorator and used 2 with's.  I decided to still with the decorators, as they're used pretty consistently in this class.